ctest-next/src/generator.rs

```diff
@@ -76,9 +99,183 @@ impl TestGenerator {
         self
     }
 
+    /// Skipped item names are printed to `stderr` if `skip` is `true`.
+    pub fn verbose_skip(&mut self, skip: bool) -> &mut Self {
+        self.verbose_skip = skip;
+        self
+    }
+
+    /// Is volatile?
+    ///
+    /// The closure given takes a `VolatileKind` denoting a particular item that
+    /// could be volatile, and returns whether this is the case.
+    pub fn volatile_item(&mut self, f: impl Fn(VolatileItemKind) -> bool + 'static) -> &mut Self {
+        self.volatile_item = Some(Box::new(f));
+        self
+    }
+
+    /// Is argument of function an array?
+    ///
+    /// The closure denotes whether particular argument of a function is an array.
+    pub fn array_arg<F>(
+        &mut self,
+        f: impl Fn(crate::Fn, Parameter) -> bool + 'static,
+    ) -> &mut Self {
+        self.array_arg = Some(Box::new(f));
+        self
+    }
+
+    /// Configures whether the tests for a struct are emitted.
```

Comment:
To demo usage, could you please add a doc example to all these added methods? They probably have to be `no_run` like ctest is